common/flatpak-dir.c

```diff
@@ -7940,6 +7944,10 @@ flatpak_dir_install (FlatpakDir          *self,
           if (!child_repo_ensure_summary (child_repo, state, cancellable, error))
             return FALSE;
 
+          if (dup_socket &&
+              shutdown (dup_socket, SHUT_RDWR) == -1)
```

Comment:
I don't think this code belongs here. It is the system helper which needs to make sure the writer has been closed before pulling from the temp repo into the system repo. It can't trust the client to do it. The client should unmount the FUSE bit, and tell the system helper to pull from the real directory.